### DISCUSSION TITLE
Inconsistent column data: Malformed ObjectID

### DISCUSSION THREAD

**Discussion by happyrw** [prisma/prisma#22975] [Q&A]
### Question

Hey guys, I have an issue in signIn using Google and GitHub providers I get this kind of error 

```
[auth][error] AdapterError: Read more at https://errors.authjs.dev#adaptererror
[auth][cause]: PrismaClientKnownRequestError:
Invalid `prisma.user.create()` invocation:

Inconsistent column data: Malformed ObjectID: provided hex string representation must be exactly 12 bytes, instead got: "143003179", length 9 for the field 'id'.
```
```js
datasource db {
  provider  = "mongodb"
  url  	    = env("DATABASE_URL")
}

enum UserRole {
  ADMIN
  USER
}

model User {
  id            String    @id @default(auto()) @map("_id") @db.ObjectId
  name          String?
  email         String?   @unique
  password      String?
  emailVerified DateTime?
  image         String?
  role          UserRole @default(USER)
  accounts      Account[]
}
```

and My credentials work platy fine.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @happyrw 👋 

Thank you for raising this question.

The error message suggests that the `id` value being provided is not a valid `ObjectID` for MongoDB, which requires a 12-byte hex string.

Check the source of the `id` values being used in your `prisma.user.create()` invocations. If these` id` values are being generated or provided by the Google and GitHub sign-in processes, you might need to adjust your logic to ensure that valid 12-byte hex strings are being used.

You may also want to share the actual query and the surrounding context that is generating this error.

**Reply by happyrw** [reply]
[GOOGLE & GITHUB LOGIN.docx](https://github.com/prisma/prisma/files/14199866/GOOGLE.GITHUB.LOGIN.docx)


All my codes related to GOOGLE & GITHUB are in the file provided. Take a look and help me my issue please. 
Thank you for your time sir !

**Reply by happyrw** [reply]
@ludralph Hello 👋

**Reply by alexfelker12** [reply]
@happyrw Since MongoDB expects a 12 bytes hex string representation for an ObjectID, "143003179" is not valid. Weirdly, somwhere in the authorization process, the id from the user gets converted to a number in the user object from next-auth.

So this line: `if (token?.sub) session.user.id = token.sub;` would set a different id ("143003179") than the actual user id in your db.

A solution would be to get the User manually from the db and asign the id from the user-entry to `token.id` and then setting `user.id` in the session object equal the `token.id`.

The code would look something like this.
```ts
  callbacks: {
    // other callbacks...
    jwt: async function({ token, profile }) {
      if (profile) {
        const user = await prisma.user.findUnique({
          where: {
            email: profile.email,
          },
        })
        if (!user) {
          throw new Error('No user found')
        }
        token.id = user.id
      }
      return token
    },
    session: function({ session, token }) {
      return {
        ...session,
        user: {
          ...session.user,
          id: token.id
        }
      }
    },
  },
```

This gives me the correct Id in the session object. I hope this helps you!

P.S.: If somebody knows what exactly happens to the id in the next-auth user object when using MongoDB and how to get the correct hex string back I would gladly like to know. :)

**Reply by ludralph** [reply]
Hi @happyrw 

Did the suggestion by Alex help resolve your issue? If, no, can you please share your coe preferably in a github repo instead of a docx file?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
